src/main/java/io/lsdconsulting/lsd/distributed/generator/config/LibraryConfig.java

```diff
@@ -0,0 +1,70 @@
+package io.lsdconsulting.lsd.distributed.generator.config;
+
+import com.lsd.IdGenerator;
+import com.lsd.LsdContext;
+import io.lsdconsulting.lsd.distributed.access.repository.InterceptedDocumentRepository;
+import io.lsdconsulting.lsd.distributed.generator.diagram.InteractionGenerator;
+import io.lsdconsulting.lsd.distributed.generator.diagram.LsdLogger;
+import io.lsdconsulting.lsd.distributed.generator.diagram.data.InteractionDataGenerator;
+import io.lsdconsulting.lsd.distributed.generator.diagram.event.ConsumeMessageBuilder;
+import io.lsdconsulting.lsd.distributed.generator.diagram.event.EventBuilderMap;
+import io.lsdconsulting.lsd.distributed.generator.diagram.event.MessageBuilder;
+import io.lsdconsulting.lsd.distributed.generator.diagram.event.SynchronousResponseBuilder;
+import io.lsdconsulting.lsd.distributed.generator.diagram.label.LabelGeneratorMap;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConditionalOnProperty(name = "lsd.dist.db.connectionString")
+public class LibraryConfig {
```

Comment:
We have solved it for the `save` operation by swallowing `MongoException`s. Maybe an easier approach would be to do the same for `InterceptedDocumentMongoRepository#prepareMongoClient` method? It would not require switching a boolean property, would it? We could make sure that if the MongoClient was not created successfully, any operation on the repository would be ineffective. The app would start regardless of whether there is a DB or not and the captors would just not save anything to the DB.

The question is, do we actually want this to be so silent?

Reply:
@nickmcdowall please check out the latest commits in the `lsd-distributed-mongodb-access` project and let me know if those changes fix the problem sufficiently.